### PR TITLE
Backport #1492 to v0.12

### PR DESCRIPTION
**Pull request by cosmo0920**

### CODE REVIEW

File: lib/fluent/parser.rb [hunk 1]
```diff
@@ -549,6 +553,26 @@ def configure(conf)
         super
 
         @regexp = @with_priority ? REGEXP_WITH_PRI : REGEXP
```
Comment (repeatedly) [inline lib/fluent/parser.rb]: Can we remove this line?

File: lib/fluent/parser.rb [hunk 2]
```diff
@@ -557,6 +580,23 @@ def patterns
       end
 
       def parse(text)
+        # This is overwritten in configure
+      end
+
+      def parse_auto(text, &block)
+        if @message_format == :auto
```
Comment (repeatedly) [inline lib/fluent/parser.rb]: We can remove this condition because `parse_auto` is not called directly.

**Comment by repeatedly** [comment]
Commented. I will merged this patch after applied reviews.

File: lib/fluent/parser.rb (hunk 2 above)
Comment (cosmo0920) [inline lib/fluent/parser.rb]: Oh! I've overlooked this. :confounded:

File: lib/fluent/parser.rb [hunk 3]
```diff
@@ -548,7 +552,26 @@ def initialize
...
+                  when :rfc3164
+                    class << self
+                      alias_method :parse, :parse_plain
+                    end
+                    @with_priority ? REGEXP_WITH_PRI : REGEXP
+                  when :rfc5424
+                    class << self
+                      alias_method :parse, :parse_plain
+                    end
+                    REGEXP_RFC5424
```
Comment (repeatedly) [inline lib/fluent/parser.rb]: Assigninig `@time_format = @rfc5424_time_format` is needed?

**Comment by repeatedly** [comment]
I re-checked the code and I have one question. Commented it.

File: lib/fluent/parser.rb (hunk 3 above)
Comment (cosmo0920) [inline lib/fluent/parser.rb]: Ah! Your suggestion means that we should set `@time_format` in the following situation?

```ruby
@time_format = @rfc5424_time_format unless @time_format
```

It is reasonable for me.

Comment (repeatedly) [inline lib/fluent/parser.rb]: Yes, but time_format has default value.
So there is two approaches:

- Change time_format's default to `nil` and set each default time_format in `message_format` condition
- Use conf.has_key?('time_format')

Comment (cosmo0920) [inline lib/fluent/parser.rb]: I want to choose `conf.has_key?('time_format')` approach.

**Comment by repeatedly** [comment]
Thanks!
Could you add test for only `message_format rfc5424` configuration?

**Comment by cosmo0920** [comment]
> Could you add test for only message_format rfc5424 configuration?

I've added in https://github.com/fluent/fluentd/pull/1495/commits/223d249d212563809b5752a90c2c10b55b992075.

**Comment by repeatedly** [comment]
Thanks!

**Pull request merged**